### DISCUSSION TITLE
Overview and/or documentation on Remote Instance / Remote Client?

### DISCUSSION THREAD

**Discussion by willnationsdev** [OrchardCMS/OrchardCore#8768] [Q&A] [answered]
I can't seem to find any information on what the difference between Remote Instances and Remote Clients are, how you generate the proper keys for them, how you set them up / create them so that they work properly, etc. I see no mention of them in the docs, in any discussions or issues, or even in the original Orchard docs.

I may just be missing the blindly obvious, but could someone give me an idea of how to use these? I have deployment plans defined, but, for now, I've had to manually export a zip of the recipe file and then do package import in the target environment in order to perform the migrations I want. Would prefer to configure the process once and automate it with the click of a button.

**Comment by deanmarcussen** [comment, marked as answer]
On your production server create a `Remote Client`
- Client Name = `prod`
- ApiKey = `whatever you want`(it's data protected, so is just a value of your choice)
- Take the Url to use.

On your staging server create a `Remote Instance`
- Call it whatever you want (prod)
- Use the Client Name = `'prod'
- ApiKey = `whatever you set on the previous step`
- Url = the url recorded from the Remote Client

On your staging server you will now have a prod target 

Repeat the same procedure if you want to setup a target from prod -> staging on the prod server.

Assuming this works ;) Please create some documentation from the steps above a you are quit right we don't have that documented anywhere.

**Reply by willnationsdev** [reply]
Great, thanks. I'll give this a try as soon as I can (though it'll be a little bit). Once I verify it works, I'll post a docs PR.

**Reply by willnationsdev** [reply]
When I tried to do this, we got an error message that read...

```
An error occurred while sending the deployment to the remote instance: "Bad Request (400)"
```

I'm not sure if we missed something else or if it has to do with how our server is configured.

**Reply by deanmarcussen** [reply]
Normally the client name or api key is mismatched when I get that error
